doc/reference/kernel/threads/index.rst

```diff
@@ -301,9 +357,22 @@ Spawning a Thread
 =================
 
 A thread is spawned by defining its stack area and its thread control block,
-and then calling :cpp:func:`k_thread_create()`. The stack area must be defined
-using :c:macro:`K_THREAD_STACK_DEFINE` to ensure it is properly set up in
-memory.
+and then calling :cpp:func:`k_thread_create()`.
+
+The stack area must be defined using :c:macro:`K_THREAD_STACK_DEFINE` or
+:c:macro:`K_KERNEL_STACK_DEFINE` to ensure it is properly set up in memory.
+
+The size parameter for the stack must be one of three values:
+
+- The original requested stack size passed to
+  ``K_THREAD_STACK`` or ``K_KERNEL_STACK`` family of stack instantiation
+  macros.
+- For a stack object defined with the ``K_THREAD_STACK`` family of
+  macros, the return value of :c:macro:`K_THREAD_STACK_SIZEOF()` for that'
+  object.
+- For a stack object defined with the ``K_KERNEL_STACK`` family of
+  macros, the return value of :c:macro:`K_KERNEL_STACK_SIZEOF()` for that
+  object.
```

Comment:
Second and third are the same

Reply:
The doxygen API documentation just says to use `K_KERNEL_STACK_SIZEOF`, maybe it is worth to add the other possibilities there as well.

Reply:
@ceolin no they aren't the same.

One discusses `K_THREAD_STACK_*` macros, the other `K_KERNEL_STACK`

Reply:
What shame, just read the same thing. Sorry.